src/components/registrationForm/RegistrationForm.tsx

```diff
@@ -0,0 +1,218 @@
+import React, { useEffect, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import './RegistrationForm.css';
+
+export default function RegistrationForm() {
+	const {
+		register,
+		getValues,
+		handleSubmit,
+		formState: { errors, isValid },
+		setError,
+		clearErrors,
+	} = useForm({
+		defaultValues: {
+			login: '',
+			email: '',
+			password: '',
+			confirmedPassword: '',
+		},
+		mode: 'onChange',
+	});
+
+	const [isFormValid, setFormValid] = useState(false);
+	const [isLoginValid, setLoginValid] = useState(false);
+	const [isEmailValid, setEmailValid] = useState(false);
+	const [isPasswordValid, setPasswordValid] = useState(false);
+	const [isConfirmedPasswordValid, setConfirmedPasswordValid] =
+		useState(false);
+
+	useEffect(() => {
+		setFormValid(isValid);
+	}, [isValid]);
+
+	const [serverError, setServerError] = useState<string | null>();
+
+	const submitForm = async (data: any) => {
+		try {
+			const response = await fetch(
+				'https://localhost:7238/api/users/signup',
+				{
+					method: 'POST',
+					headers: { 'Content-Type': 'application/json' },
+					body: JSON.stringify(data),
+				},
+			)
+				.then((res) => {
+					if (res.ok) {
+						alert("You're registered!");
+						setServerError(null);
+					} else {
+						return res.json();
+					}
+				})
+				.then((data) => {
+					switch (data.code) {
+						case 400:
+							setServerError(
+								`${data.message} [${data.code} Bad request]`,
+							);
+							break;
+						case 401:
+							setServerError(
+								`${data.message} [${data.code} Unauthorized]`,
+							);
+							break;
+						case 409:
+							setServerError(
+								`${data.message} [${data.code} Conflict]`,
+							);
+							break;
+					}
+				});
+
+			// TODO: handle error handling xD
+
+			// if (!response.ok) {
+			// 	switch (response.status) {
+			// 		case 400:
+			// 			setServerError(`${response.status}: Bad request.`);
+			// 			//console.log(serverError);
+			// 			break;
+			// 		case 401:
+			// 			setServerError(`${response.status}: Unauthorized.`);
+			// 			//console.log(serverError);
+			// 			break;
+			// 		case 409:
+			// 			setServerError(`${response.status}: Data in use.`);
+			// 			//console.log(serverError);
+			// 			break;
+			// 	}
+			// } else {
+			// 	//console.log(response.status);
+			// 	alert('Thank you for registration!');
+			// 	setServerError(null);
+			// }
+		} catch (err) {
+			//setServerError(`${err}`);
+		}
+	};
```

Comment:
Can be extracted to separate files:
Fetch (and any get, post, update, delete) into `@utils/fetchApi.ts` with props like url, states and setters,
Error handling into `@utils/apiErrorHandling.ts` passing in response code as prop,
finally submitForm imported `import { useSubmitForm } from '@hooks/useSubmitForm.ts' ` accepting required props that will be passed into utility functions